src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -660,11 +660,24 @@ private BoundCollectionExpression ConvertCollectionExpression(
 
             if (collectionTypeKind is CollectionExpressionTypeKind.ImplementsIEnumerable)
             {
+                if (targetType is NamedTypeSymbol namedType &&
+                    HasParamsCollectionTypeInProgress(namedType, out NamedTypeSymbol? inProgress, out MethodSymbol? inProgressConstructor))
+                {
+                    Debug.Assert(inProgressConstructor is not null);
+                    diagnostics.Add(ErrorCode.ERR_ParamsCollectionInfiniteChainOfConstructorCalls, syntax, inProgress, inProgressConstructor.OriginalDefinition);
+                    return BindCollectionExpressionForErrorRecovery(node, namedType, inConversion: true, diagnostics);
+                }
+
                 implicitReceiver = new BoundObjectOrCollectionValuePlaceholder(syntax, isNewInstance: true, targetType) { WasCompilerGenerated = true };
-                collectionCreation = BindCollectionExpressionConstructor(syntax, targetType, diagnostics);
+                collectionCreation = BindCollectionExpressionConstructor(syntax, targetType, constructor, diagnostics);
                 Debug.Assert((collectionCreation is BoundNewT && !isExpanded && constructor is null) ||
                              (collectionCreation is BoundObjectCreationExpression creation && creation.Expanded == isExpanded && creation.Constructor == constructor));
 
+                if (collectionCreation.HasErrors)
```

Comment:
>collectionCreation.HasErrors [](http://example.com/codeflow?start=20&length=28)

Just curious: Why is this check needed rather than continuing and binding the elements?

Reply:
It's not needed, but it prevents cascading diagnostics. For example, in call like `Test(2, 3);`, the cycle error would be reported on `Test(2, 3)`, `2`, and `3`.